src/EditorFeatures/Core/SymbolSearch/SymbolSearchUpdateEngine.Update.cs

```diff
@@ -222,25 +223,63 @@ private string ConvertToFileName(string source)
 
             private async Task<TimeSpan> DownloadFullDatabaseAsync()
             {
+                try
+                {
+                    var title = string.Format(WorkspaceDesktopResources.Downloading_IntelliSense_index_for_0, _source);
+                    await _service._progressService.OnDownloadFullDatabaseStartedAsync(title).ConfigureAwait(false);
+
+                    var (succeeded, delay) = await DownloadFullDatabaseWorkerAsync().ConfigureAwait(false);
+                    if (succeeded)
+                    {
+                        await _service._progressService.OnDownloadFullDatabaseSucceededAsync().ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        await _service._progressService.OnDownloadFullDatabaseFailedAsync(
+                            WorkspaceDesktopResources.Downloading_index_failed).ConfigureAwait(false);
+                    }
+
+                    return delay;
+                }
+                catch (OperationCanceledException)
+                {
+                    await _service._progressService.OnDownloadFullDatabaseCanceledAsync().ConfigureAwait(false);
+                    throw;
+                }
+                catch (Exception e)
+                {
+                    var message = string.Format(
+                        WorkspaceDesktopResources.Downloading_index_failed_0,
+                        "\r\n" + e.ToString());
+                    await _service._progressService.OnDownloadFullDatabaseFailedAsync(message).ConfigureAwait(false);
```

Comment:
Do we know if we're going to be throwing a bunch of unlocalized exception messages here?